From 517096713fb1ae695c4720a479d64aa09497c9f5 Mon Sep 17 00:00:00 2001
From: Aaradhana Sahu <quic_aarasahu@quicinc.com>
Date: Thu, 22 Dec 2022 12:12:43 +0530
Subject: [PATCH] ath12k: fix firmware assert issue for Tx test case in FTM

Currently, we are converting hardware cookies for FTM mode
but firmware is not handling such type of cookie conversion,
so it's trigger the firmware assert.

Fix this issue by disabling hardware cookie conversion in
FTM mode.

Signed-off-by: Aaradhana Sahu <quic_aarasahu@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -1306,6 +1306,9 @@ void ath12k_dp_cc_config(struct ath12k_b
 	u32 wbm_base = HAL_SEQ_WCSS_UMAC_WBM_REG;
 	u32 val = 0;
 
+	if (ath12k_ftm_mode)
+		return;
+
 	ath12k_hif_write32(ab, reo_base + HAL_REO1_SW_COOKIE_CFG0(ab), cmem_base);
 
 	val |= u32_encode_bits(ATH12K_CMEM_ADDR_MSB,
